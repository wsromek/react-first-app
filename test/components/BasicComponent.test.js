import React from 'react';
import BasicComponent from '../../src/client/app/components/BasicComponent.jsx';
import renderer from 'react-test-renderer';

test('BasicComponent increments counter on click ', () => {
    const component = renderer.create(
        <BasicComponent />
    );

    let tree = component.toJSON();
    let likeButton = tree.children[2].children[0];

    expect(tree).toMatchSnapshot();

    likeButton.props.onClick();

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
