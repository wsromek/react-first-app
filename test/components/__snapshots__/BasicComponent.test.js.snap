// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicComponent increments counter on click  1`] = `
<div>
  Likes : 
  <span>
    0
  </span>
  <div>
    <button
      onClick={[Function]}
    >
      Like Me
    </button>
  </div>
</div>
`;

exports[`BasicComponent increments counter on click  2`] = `
<div>
  Likes : 
  <span>
    1
  </span>
  <div>
    <button
      onClick={[Function]}
    >
      Like Me
    </button>
  </div>
</div>
`;

exports[`BasicComponent increments counter on click  3`] = `
<div>
  Likes : 
  <span>
    1
  </span>
  <div>
    <button
      onClick={[Function]}
    >
      Like Me
    </button>
  </div>
</div>
`;
